pkl-core/src/main/java/org/pkl/core/packages/PackageResolvers.java

```diff
@@ -490,7 +490,8 @@ private Path getMetadataPath(
       if (Files.exists(cachePath)) {
         return cachePath;
       }
-      var tmpPath = tmpDir.resolve(metadataRelativePath);
+      Files.createDirectories(tmpDir);
+      var tmpPath = Files.createTempFile(tmpDir, null, ".json");
```

Comment:
Not blocking, but for `/tmp` hygiene, an informative prefix would be good. I'd just use the package name & version.